### PR TITLE
Added the option to restore recently closed tabs

### LINKED ISSUE #2117: Add `ctrl + shift + t` to restore a recently closed tab

### PR DESCRIPTION
**Pull request by d2dyno1**
This PR enables users to restore recently closed tab. Furthermore, the changes in this PR have been made in mind of future improvements - to allow displaying entire history of recently closed tabs. This PR also brings small optimizations, code flexibility and overall code cleanup related to tabs.

Please leave your opinions on this PR.

Resolves #2117

### CODE REVIEW

**Comment by d2dyno1** [comment]
I've noticed an issue with it but I'm not entirely sure how to fix it. The problem lies with invoking the key combination, in `switch` statement there's an additional parameter called `tabInstance` of type `bool`. How can I get it so it's ignored in the `ctrl+shift+t` case?

@yair100 

[**Solved**]

File: Files/Views/ModernShellPage.xaml.cs [hunk 1]
```diff
@@ -891,6 +891,16 @@ private async void KeyboardAccelerator_Invoked(KeyboardAccelerator sender, Keybo
                     await storageHistoryHelpers.TryRedo();
                     break;
 
+                case (true, true, false, true, VirtualKey.T): // ctrl + shif + t, restore recently closed tab
```
Comment (d2dyno1) [inline Files/Views/ModernShellPage.xaml.cs:894]: Here

Comment (yaichenbaum) [reply Files/Views/ModernShellPage.xaml.cs:894]: What does `tabInstance` represent?

Comment (d2dyno1) [reply Files/Views/ModernShellPage.xaml.cs]: Tab instance is true whether `CurrentPageType` is `GenericFileBrowser` or `GenericFileBrowser`

Comment (d2dyno1) [reply Files/Views/ModernShellPage.xaml.cs]: Ok, I solved it by replacing the bool with discard `_` symbol

**Review by yair100 — DISMISSED** [review]

**Comment by yaichenbaum** [comment]
@d2dyno1 Should we add the option to reopen a closed tab to the horizontal tab view context menu?

**Comment by d2dyno1** [comment]
@yair100 Yeah, why not?

**Comment by yaichenbaum** [comment]
@d2dyno1 I found two issues,
- `Ctrl + shift + t` crashes the app if you try the shortcut before any tabs were closed.
- `Ctrl + shift + t` crashes the app if the most recently closed tab was the new tab page.

**Comment by d2dyno1** [comment]
@yair100 I've fixed the first issue. I couldn't reproduce the second one. I don't expect it to crash on home page since I've added a null check there (in `MainPage.AddNewTabByPathAsync()`)

**Review by yair100 — APPROVED** [review]

**Pull request merged**